File: java/src/main/java/vsdk/toolkit/material/Material.java
package vsdk.toolkit.material;

// Emittance distribution function

// Reflection and transmission together

// True for 1-sided surface, false for 2-sided, see mgf docs

// Material name

public class Material {
    private PhongEmittanceDistributionFunction edf;
    private PhongBidirectionalScatteringDistributionFunction bsdf;
    private boolean sided;
    private String name;

    public Material(
        String inName,
        PhongEmittanceDistributionFunction inEdf,
        PhongBidirectionalScatteringDistributionFunction inBsdf,
        boolean inSided) {
        name = inName == null ? "" : inName;
        sided = inSided;
        edf = inEdf;
        bsdf = inBsdf;
    }

    public PhongEmittanceDistributionFunction getEdf() {
        return edf;
    }

    public PhongBidirectionalScatteringDistributionFunction getBsdf() {
        return bsdf;
    }

    public boolean isSided() {
        return sided;
    }

    public String getName() {
        return name;
    }
}

File: typescript/base/src/main/vsdk/toolkit/material/Material.ts
import { PhongBidirectionalScatteringDistributionFunction } from "./PhongBidirectionalScatteringDistributionFunction";
import { PhongEmittanceDistributionFunction } from "./PhongEmittanceDistributionFunction";

export class Material {
  private readonly edf: PhongEmittanceDistributionFunction;
  private readonly bsdf: PhongBidirectionalScatteringDistributionFunction;
  private readonly sided: boolean;
  private readonly name: string;

  public constructor(
    inName: string | null,
    inEdf: PhongEmittanceDistributionFunction,
    inBsdf: PhongBidirectionalScatteringDistributionFunction,
    inSided: boolean
  ) {
    this.name = inName ?? "";
    this.sided = inSided;
    this.edf = inEdf;
    this.bsdf = inBsdf;
  }

  public getEdf(): PhongEmittanceDistributionFunction {
    return this.edf;
  }

  public getBsdf(): PhongBidirectionalScatteringDistributionFunction {
    return this.bsdf;
  }

  public isSided(): boolean {
    return this.sided;
  }

  public getName(): string {
    return this.name;
  }
}
